tests/models/abiotic/test_conductivities.py

```diff
@@ -52,43 +58,42 @@ def test_interpolate_along_heights(dummy_climate_data, fixture_core_components):
         interpolate_along_heights,
     )
 
-    layer_heights = dummy_climate_data["layer_heights"]
-    atmos_index = np.logical_not(
-        fixture_core_components.layer_structure.role_indices_bool["all_soil"]
-    )
-    atmosphere_layers = layer_heights[atmos_index]
+    lyr_str = fixture_core_components.layer_structure
+
+    layer_heights = dummy_climate_data["layer_heights"].to_numpy()
+
     result = interpolate_along_heights(
-        start_height=layer_heights[-3].to_numpy(),
-        end_height=layer_heights[0].to_numpy(),
-        target_heights=(layer_heights[atmosphere_layers.indexes].to_numpy()),
+        start_height=layer_heights[lyr_str.index_surface],
+        end_height=layer_heights[lyr_str.index_above],
+        target_heights=layer_heights[lyr_str.index_filled_atmosphere],
         start_value=50.0,
         end_value=20.0,
     )
 
     # Get layer structure and reduce to only atmospheric layers
-    exp_result = fixture_core_components.layer_structure.from_template()
-    exp_result.T[..., [0, 1, 2, 3, 11]] = [
-        20.0,
-        21.88087774,
-        31.28526646,
-        40.68965517,
-        50.0,
-    ]
-    exp_result = exp_result[atmos_index]
+    exp_result = lyr_str.from_template()
+    exp_result[lyr_str.index_filled_atmosphere] = np.array(
+        [20.0, 21.88087774, 31.28526646, 40.68965517, 50.0]
+    )[:, None]
+    exp_result = exp_result[lyr_str.index_filled_atmosphere]
 
     np.testing.assert_allclose(result, exp_result, rtol=1e-04, atol=1e-04)
 
 
 def test_interpolate_along_heights_arrays(fixture_core_components, dummy_climate_data):
     """Test linear interpolation along heights with arrays of boundary values."""
 
+    # TODO - I don't think this differs from the test above.
+
```

Comment:
The difference is that in the first test, the boundaries are floats, in the second one they are arrays. Could probably drop the first one

Reply:
OK - no. I see the difference.